### PR TITLE
Remove deprecated property calls

### PR DESCRIPTION
**Pull request by grasskin**
Not sure why the loss is different even if the results are equally good at least in the confusion matrix. I would have expected the loss calculation not to change. It is still learning.

### CODE REVIEW

**Comment by grasskin** [comment]
We might be using the Mean metric by default, let me make that more explicit.

File: examples/vision/metric_learning.py [hunk 1]
```diff
@@ -192,14 +192,17 @@ def train_step(self, data):
             # pairs, to be high. This loss will move embeddings for the
             # anchor/positive pairs together and move all other pairs apart.
             sparse_labels = keras.ops.arange(num_classes)
-            loss = self.compiled_loss(sparse_labels, similarities)
+            loss = self.compute_loss(x=None, y=sparse_labels, y_pred=similarities)
```
Comment (fchollet) [inline examples/vision/metric_learning.py:195]: Tip: you can omit `x=None` here. Not important though

File: examples/vision/metric_learning.py [hunk 2]
```diff
@@ -192,14 +192,17 @@ def train_step(self, data):
...
 
         # Calculate gradients and apply via optimizer.
         gradients = tape.gradient(loss, self.trainable_variables)
         self.optimizer.apply_gradients(zip(gradients, self.trainable_variables))
 
         # Update and return metrics (specifically the one for the loss value).
-        self.compiled_metrics.update_state(sparse_labels, similarities)
+        for metric in self.metrics:
+            # Calling `self.compile` will by default add a `keras.metrics.Mean` metric
+            metric.update_state(similarities)
```
Comment (fchollet) [inline examples/vision/metric_learning.py:204]: This will not generalize if there are more metrics passed to compile though -- prefer reusing the standard snippet from https://keras.io/keras_core/guides/custom_train_step_in_tensorflow/#a-first-simple-example

File: examples/vision/metric_learning.py (hunk 1 above)
Comment (grasskin) [reply examples/vision/metric_learning.py]: Removed!

File: examples/vision/metric_learning.py (hunk 2 above)
Comment (grasskin) [reply examples/vision/metric_learning.py]: Switched to standard snippet - at some later point we should circle back to how metrics are being dealt with because there's some weirdness in how we're handling Sum/Mean.

**Review by fchollet — APPROVED** [review]
LGTM, thank you
